pyo3-build-config/src/lib.rs

```diff
@@ -70,7 +71,7 @@ pub fn get() -> &'static InterpreterConfig {
             InterpreterConfig::from_reader(Cursor::new(CONFIG_FILE))
         } else if !ABI3_CONFIG.is_empty() {
             Ok(abi3_config())
-        } else if impl_::cross_compile_env_vars().any() {
+        } else if Path::new(DEFAULT_CROSS_COMPILE_CONFIG_PATH).exists() {
```

Comment:
This does make sense to me, but I am wary whether this would pick up stale configuration files when the environment variables change and a project is rebuilt?

Reply:
This is a tricky question. I think this file is created by the `pyo3-ffi` build script, which should re-run when the rustc target or the PyO3 env vars change, but I'm not 100% sure.

`cargo build --target X` worked for me without running `cargo clean` before changing `X`.
Perhaps there is a better way to test this.

@davidhewitt What do you think about this approach to detecting whether we are really cross-compiling?

Reply:
> which should re-run when the rustc target or the PyO3 env vars change, but I'm not 100% sure.

I was thinking mainly about the case when switching from a cross build to a non-cross build as the build script probably does not get out of its way to delete a cross compilation config written by a prior run.

> cargo build --target X worked for me without running cargo clean before changing X. 

I would expect this to work as the file name is the same and the config would be overwritten.

I think https://github.com/PyO3/pyo3/pull/2092 would actually be a nice solution to this as the `DEP_*` environment variables are basically meant for this. But I guess we want to keep the text files around in any case?

Reply:
Yes the `DEP_` environment variables sadly don't work for the proc macro crate so we need this backup mechanism anyway.

Reply:
> > which should re-run when the rustc target or the PyO3 env vars change, but I'm not 100% sure.
> 
> I was thinking mainly about the case when switching from a cross build to a non-cross build as the build script probably does not get out of its way to delete a cross compilation config written by a prior run.

But, isn't `pyo3-cross-compile-config.txt` created in the directory from the `OUT_DIR` Cargo variable, which should be pointing to somewhere in `target/<triple>/deps`? I think config file path should not collide at least as far as `--target` arguments are concerned.

Reply:
Actually, I was completely wrong about `pyo3-cross-compile-config.txt` location. It is created in the host build script directory because `pyo3-build-config` is always compiled for the host arch.

I'm using a different approach in https://github.com/PyO3/pyo3/pull/2241/commits/fd6a68e6798c31ee74563f6d29d6560ce38d94f3.

Reply:
This seems like a nice clean-up to me! Thanks!